app/Models/Asset.php

```diff
@@ -255,11 +255,15 @@ public function availableForCheckout()
      * @since [v3.0]
      * @return boolean
      */
-    public function checkOut($target, $admin = null, $checkout_at = null, $expected_checkin = null, $note = null, $name = null, $location = null)
+    public function checkOut($target, $admin = null, $status_id = null, $checkout_at = null, $expected_checkin = null, $note = null, $name = null, $location = null)
```

Comment:
This does not seem like the right place to put this - simply updating the status (while making sure that the echoing effects are observed, such as a user location change which we echo out to the asset location) is handled.

Any method with this many functions should make you wonder if there's not a smarter way to handle this.